lib/ProviderAws.js

```diff
@@ -91,86 +91,79 @@ class ServerlessProviderAws {
+    
     stage = stage ? stage.toUpperCase() : null;
 
-    if (stage && process.env['AWS_PROFILE_' + stage]) {
-
-      // Profile w/ Stage Suffix
-      let profile = process.env['AWS_PROFILE_' + stage];
-      profile     = (this._config.profilePrefix ? this._config.profilePrefix + '_' + profile : profile).toLowerCase();
-      credentials = this.getProfile(profile);
-
-    } else if (process.env['AWS_PROFILE'] || process.env['SERVERLESS_ADMIN_AWS_PROFILE']) {
-
-      // Profile Plain
-      let profile = process.env['AWS_PROFILE'] || process.env['SERVERLESS_ADMIN_AWS_PROFILE'];
-      profile     = (this._config.profilePrefix ? this._config.profilePrefix + '_' + profile : profile).toLowerCase();
-      credentials = this.getProfile(profile);
-
-    } else if (process.env['AWS_ACCESS_KEY_ID_'  + stage] && process.env['AWS_SECRET_ACCESS_KEY_'  + stage]) {
-
-      // Access Keys w/ Stage Suffix
-      credentials = {
-        accessKeyId:     process.env['AWS_ACCESS_KEY_ID_' + stage],
-        secretAccessKey: process.env['AWS_SECRET_ACCESS_KEY_' + stage]
-      };
-
-    } else if ((process.env['AWS_ACCESS_KEY_ID'] && process.env['AWS_SECRET_ACCESS_KEY'])
-      || process.env['SERVERLESS_ADMIN_AWS_ACCESS_KEY_ID'] && process.env['SERVERLESS_ADMIN_AWS_SECRET_ACCESS_KEY']) {
-
-      // Access Keys Plain
-      credentials = {
-        accessKeyId:     process.env['AWS_ACCESS_KEY_ID'] || process.env['SERVERLESS_ADMIN_AWS_ACCESS_KEY_ID'],
-        secretAccessKey: process.env['AWS_SECRET_ACCESS_KEY'] || process.env['SERVERLESS_ADMIN_AWS_SECRET_ACCESS_KEY']
-      };
-
-    } else if (process.env['AWS_SESSION_TOKEN_' + stage]) {
-
-      // Session Token w/ Stage Suffix
-      credentials = {
-        sessionToken:    process.env['AWS_SESSION_TOKEN_' + stage]
-      };
-
-    } else if (process.env['AWS_SESSION_TOKEN']) {
-
-      // Session Token Plain
-      credentials = {
-        sessionToken:    process.env['AWS_SESSION_TOKEN']
-      };
-    } else if (this._S.config.awsAdminKeyId) {
-
-      // Access Keys from the config
-      credentials = {
-        accessKeyId:     this._S.config.awsAdminKeyId,
-        secretAccessKey: this._S.config.awsAdminSecretKey
-      };
-    }
-
-    if (!credentials) {
+    // implicitly already in the config...
+    this.addConfigurationCredentials(credentials, this._S.config);                       // use the given configuration credentials if they are the only available credentials.
+    // first from environment
+    this.addEnvironmentCredentials(credentials, 'AWS');                      // allow for Amazon standard credential environment variable prefix.
+    this.addEnvironmentCredentials(credentials, 'SERVERLESS_ADMIN_AWS');          // but override with more specific credentials if these are also provided.
+    this.addEnvironmentCredentials(credentials, 'AWS' + stage);                  // and also override these with the Amazon standard *stage specific* credential environment variable prefix.
```

Comment:
I think this should be `this.addEnvironmentCredentials(credentials, 'AWS_' + stage);` (Missing underscore), same on the next line.